pkg/provider/api.go

```diff
@@ -399,19 +395,22 @@ func (ap *ApiProvider) RefreshUsers(ctx context.Context) error {
 		optionLimit  = slack.GetUsersOptionLimit(1000)
 	)
 
-	// Get team ID for cache partitioning
+	// Get team ID and user ID for cache partitioning
 	authResp, err := ap.client.AuthTest()
 	if err != nil {
 		ap.logger.Error("Failed to get auth test for team ID", zap.Error(err))
 		return err
 	}
 	teamID := authResp.TeamID
-	if teamID == "" {
-		ap.logger.Warn("Team ID is empty, skipping Redis cache operations")
+	userID := authResp.UserID
+	if teamID == "" || userID == "" {
+		ap.logger.Warn("Team ID or User ID is empty, skipping Redis cache operations",
+			zap.String("team_id", teamID),
+			zap.String("user_id", userID))
 	}
 
 	// Try to load from Redis cache first
-	redisClient, err := ap.getRedisClient(teamID)
+	redisClient, err := ap.getRedisClient(teamID, userID)
 	if err != nil {
 		ap.logger.Error("Failed to create Redis client", zap.Error(err))
 	} else if redisClient != nil {
```

Comment:
Correct me if I am wrong but we create a new redis client connection each time but don't close it. Should we close it in this else if like 
```defer redisClient.Close() ```

Just trying to avoid any connection leak.

Reply:
good catch fixed